src/core/shape_inference/include/deformable_psroi_pooling_shape_inference.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include "openvino/op/deformable_psroi_pooling.hpp"
+
+namespace ov {
+namespace op {
+namespace v1 {
+
+template <class TShape>
+std::vector<TShape> shape_infer(const DeformablePSROIPooling* op, const std::vector<TShape>& input_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() >= 2);
```

Comment:
Change to check size is 2 or 3 as more than 3 is also not correct.

Reply:
`>=2` is like a minimum requirement, the algorithm can still works fine for more than 3 inputs, just ignoring the remaining.
Anyway updated to check `2 || 3`.